src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -267,34 +306,170 @@ private ValueTask<(HttpConnection, HttpResponseMessage)> GetConnectionAsync(Http
             }
         }
 
+        private async ValueTask<(HttpConnectionBase connection, bool isNewConnection, HttpResponseMessage failureResponse)>
+            GetHttp2ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            Debug.Assert(_kind == HttpConnectionKind.Https || _kind == HttpConnectionKind.SslProxyTunnel);
```

Comment:
Any plans to allow http2 non-tls via prior knowledge? It's useful for testing and is pretty common in non-browser clients.

Reply:
Not currently, but it's probably not very difficult to do...